From 2e83856432eb7f7b0a8b883fb41232af658245de Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Fri, 25 Sep 2020 02:01:32 -0700
Subject: [PATCH 1667/1852] ASoC: xlnx: formatter: Fix warnings for UNINIT

commit 8ff5abadc8d8ec2ad62ae767c6baf22a1144061c from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the UNINIT type.

The actual warnings for below functions is:
For xlnx_formatter_pcm_open():

Event uninit_use_in_call: Using uninitialized value
"stream_data->interleaved" when calling "_dev_info"

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 sound/soc/xilinx/xlnx_formatter_pcm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/soc/xilinx/xlnx_formatter_pcm.c b/sound/soc/xilinx/xlnx_formatter_pcm.c
index ad4bf5bb9ebc..d02120310d2c 100644
--- a/sound/soc/xilinx/xlnx_formatter_pcm.c
+++ b/sound/soc/xilinx/xlnx_formatter_pcm.c
@@ -374,6 +374,8 @@ static int xlnx_formatter_pcm_open(struct snd_pcm_substream *substream)
 
 	if (!(val & data_format_mode))
 		stream_data->interleaved = true;
+	else
+		stream_data->interleaved = false;
 
 	stream_data->xfer_mode = (val & data_xfer_mode) >> data_xfer_shift;
 	stream_data->ch_limit = (val & ch_count_mask) >> ch_count_shift;
-- 
2.31.1

